src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationStateSynchronizer.cs

```diff
@@ -73,7 +77,8 @@ public void ProjectConfigurationFileChanged(ProjectConfigurationFileChangeEventA
                             return;
```

Comment:
In the line above this you're resetting the project on a `RazorFileChangeKind.Changed` which fails to deserialize. You should also be "delay updating" for resets: `EnqueueUpdateProject(projectFilePath, snapshotHandle: null)`

Then in your `UpdateAfterDelayAsync` you can see if the snapshotHandle is still `null` if it is then reset the project, otherwise update it like normal.